sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyRotationPolicy.java

```diff
@@ -3,32 +3,27 @@
 
 package com.azure.security.keyvault.keys.models;
 
-import com.azure.core.annotation.Immutable;
+import com.azure.security.keyvault.keys.implementation.models.KeyRotationPolicyAttributes;
+import com.azure.security.keyvault.keys.implementation.models.LifetimeAction;
+import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * The complete key rotation policy that belongs to a key.
  */
-@Immutable
-public final class KeyRotationPolicy extends KeyRotationPolicyProperties {
-    private final String id;
-    private final OffsetDateTime createdOn;
-    private final OffsetDateTime updatedOn;
+public final class KeyRotationPolicy {
+    @JsonProperty(value = "id", access = JsonProperty.Access.WRITE_ONLY)
+    private String id;
 
-    /**
-     * Creates an instance of {@link KeyRotationPolicy}.
-     *
-     * @param id The identifier of the {@link KeyRotationPolicy policy}.
-     * @param createdOn The {@link KeyRotationPolicy policy's} created time in UTC.
-     * @param updatedOn The {@link KeyRotationPolicy policy's} last updated time in UTC.
-     */
-    public KeyRotationPolicy(String id, OffsetDateTime createdOn, OffsetDateTime updatedOn) {
-        this.id = id;
-        this.createdOn = createdOn;
-        this.updatedOn = updatedOn;
-    }
+    private List<KeyRotationLifetimeAction> keyRotationLifetimeActions;
```

Comment:
Nit: shouldn't this match the getter/setter i.e. `lifetimeActions`? Seems that's how other fields have been.

Reply:
I think you're right, good find! :)